tidy3d/plugins/mode/mode_solver.py

```diff
@@ -308,35 +308,11 @@ def solve(self) -> ModeSolverData:
         _, solver_symmetry = self.plane.pop_axis(mode_symmetry, axis=normal_axis)
 
         # Compute and store the modes at all frequencies
-        fields = {"Ex": [], "Ey": [], "Ez": [], "Hx": [], "Hy": [], "Hz": []}
-        n_complex = []
-        for freq in self.freqs:
-            # Compute the modes
-            mode_fields, n_comp = compute_modes(
-                eps_cross=self.solver_eps(freq),
-                coords=solver_coords,
-                freq=freq,
-                mode_spec=self.mode_spec,
-                symmetry=solver_symmetry,
-            )
-            n_complex.append(n_comp)
-
-            fields_freq = {"Ex": [], "Ey": [], "Ez": [], "Hx": [], "Hy": [], "Hz": []}
-            for mode_index in range(self.mode_spec.num_modes):
-                # Get E and H fields at the current mode_index
-                ((Ex, Ey, Ez), (Hx, Hy, Hz)) = self.process_fields(mode_fields, mode_index)
-
-                # note: back in original coordinates
-                fields_mode = {"Ex": Ex, "Ey": Ey, "Ez": Ez, "Hx": Hx, "Hy": Hy, "Hz": Hz}
-                for field_name, field in fields_mode.items():
-                    fields_freq[field_name].append(np.copy(field))
-
-            for field_name, field in fields_freq.items():
-                fields[field_name].append(np.stack(field, axis=-1))
+        n_complex, fields = self._solve_all_freqs(coords=solver_coords, symmetry=solver_symmetry)
 
         # Generate the dictionary of ScalarModeFieldData for every field
         data_dict = {}
-        for field_name, field in fields.items():
+        for field_name in fields[0].keys():
```

Comment:
what's the logic behind the [0] here?

Reply:
`fields` is a List of Dicts, one for each frequency. I guess we should make sure `self.freqs` is not an empty list?

Reply:
yea or maybe check `len(fields)` to make sure it's 1.